cocotb/log.py

```diff
@@ -248,8 +248,8 @@ def format(self, record):
         msg = record.getMessage()
 
         # Need to colour each line in case coloring is applied in the message
-        msg = '\n'.join([SimColourLogFormatter.loglevel2colour[record.levelno] % line for line in msg.split('\n')])
-        level = (SimColourLogFormatter.loglevel2colour[record.levelno] %
+        msg = '\n'.join([SimColourLogFormatter.loglevel2colour.get(record.levelno,"%s") % line for line in msg.split('\n')])
```

Comment:
```suggestion
        msg = '\n'.join([SimColourLogFormatter.loglevel2colour.get(record.levelno, "%s") % line for line in msg.split('\n')])
```

Reply:
We can get this on another pass (with `black` hopefully).